File: 3.Fibonacci-Sequence/sol.go
package main

func FibonacciSequence(startSequence []int, length int) []int {
	if length <= 0 {
		return []int{}
	}

	if len(startSequence) >= length {
		return startSequence[:length]
	}

	result := make([]int, len(startSequence))
	copy(result, startSequence)

	for len(result) < length {
		last := result[len(result)-1]
		secondLast := result[len(result)-2]
		result = append(result, last+secondLast)
	}

	return result
}

File: 3.Fibonacci-Sequence/sol.py
def fibonacci_sequence(start_sequence, length):
    if len(start_sequence) >= length:
        return start_sequence[:length]
    
    result = start_sequence[:]

    while len(result) < length:
        index = len(result)
        last = result[index - 1]
        secondLast = result[index - 2]
        result.append(last + secondLast)

    return result